sos/report/plugins/__init__.py

```diff
@@ -2662,6 +2662,37 @@ def get_process_pids(self, process):
                 continue
         return pids
 
+    def filter_namespaces(self, namespaces, ns_pattern = None, ns_max = 0):
```

Comment:
Take a look at the device related commands in plugin for a reference - this should be automatically using the namespaces discovered earlier rather than having the plugins pass that same value in every time.

E.G.

```python
def get_namespaces(self, ns_pattern=None, ns_max=0):
[...]
    for ns in self.commons[namespaces][network]
```

Then in the plugins you'd use `for namespace in self.get_namespaces()` for the whole list, passing only a regex if you want to filter those down.

Reply:
I see; I tried to make the function generic in case other non-network, namespace types are implemented/used in the future.

But this approach looks much simpler for users, and it can use `get_network_namespaces()` to particularize for the network namespace type.

I'll resubmit with those changes; thanks!